libraries/aws/shadow/src/shadow.c

```diff
@@ -0,0 +1,618 @@
+    }
+
+    return returnStatus;
+}
+/*-----------------------------------------------------------*/
+
+static ShadowStatus_t extractShadowMessageType( const char * pString,
+                                                uint16_t stringLength,
+                                                ShadowMessageType_t * pMessageType )
+{
+    uint32_t index = 0U;
+    ShadowStatus_t returnStatus = SHADOW_FAIL;
+
+    /* Lookup table for Shadow message string. */
+    static const char * const pMessageStrings[ ShadowMessageTypeMaxNum ] =
+    {
+        SHADOW_OP_GET_ACCEPTED,
+        SHADOW_OP_GET_REJECTED,
+        SHADOW_OP_DELETE_ACCEPTED,
+        SHADOW_OP_DELETE_REJECTED,
+        SHADOW_OP_UPDATE_ACCEPTED,
+        SHADOW_OP_UPDATE_REJECTED,
+        SHADOW_OP_UPDATE_DOCUMENTS,
+        SHADOW_OP_UPDATE_DELTA
+    };
+
+    /* Lookup table for Shadow message string length. */
+    static const uint16_t pMessageStringsLength[ ShadowMessageTypeMaxNum ] =
+    {
+        SHADOW_OP_GET_ACCEPTED_LENGTH,
+        SHADOW_OP_GET_REJECTED_LENGTH,
+        SHADOW_OP_DELETE_ACCEPTED_LENGTH,
+        SHADOW_OP_DELETE_REJECTED_LENGTH,
+        SHADOW_OP_UPDATE_ACCEPTED_LENGTH,
+        SHADOW_OP_UPDATE_REJECTED_LENGTH,
+        SHADOW_OP_UPDATE_DOCUMENTS_LENGTH,
+        SHADOW_OP_UPDATE_DELTA_LENGTH
+    };
+
+    /* Lookup table for Shadow message types. */
+    static const ShadowMessageType_t pMessageTypes[ ShadowMessageTypeMaxNum ] =
+    {
+        ShadowMessageTypeGetAccepted ,
+        ShadowMessageTypeGetRejected,
+        ShadowMessageTypeDeleteAccepted,
+        ShadowMessageTypeDeleteRejected,
+        ShadowMessageTypeUpdateAccepted,
+        ShadowMessageTypeUpdateRejected,
+        ShadowMessageTypeUpdateDocuments,
+        ShadowMessageTypeUpdateDelta
+    };
+
+    for( ; index < ( uint32_t ) ( sizeof( pMessageStrings ) / sizeof( pMessageStrings[0] ) ); index++ )
+    {
+        returnStatus = containsSubString( pString,
+                                          stringLength,
+                                          pMessageStrings[ index ],
+                                          pMessageStringsLength[ index ] );
+
+        /* If the operation string matches, there must not be any other extra
+         * character remaining in the string. */
+        if( returnStatus == SHADOW_SUCCESS )
+        {
+            if( stringLength != pMessageStringsLength[ index ] )
+            {
+                returnStatus = SHADOW_FAIL;
+            }
+            else
+            {
+                * pMessageType = pMessageTypes[ index ];
+                break;
+            }
+        }
+    }
+
+    return returnStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+static const char * getShadowOperationString( ShadowTopicStringType_t topicType )
+{
+    const char *shadowOperationString = NULL;
+
+    switch( topicType )
+    {
+        case ShadowTopicStringTypeGet:
+        {
```

Comment:
Nit: we haven't been using braces `{}` for `switch` statements in [MQTT](https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/development/libraries/standard/mqtt/src/mqtt.c#L1944) or [HTTP](https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/development/libraries/standard/http/src/http_client.c#L2327), should we consider removing them to match the style?

Reply:
fixed it in commit [a870748](https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1081/commits/a870748a7b768e378c62babc6699ecc58115c539)

resolving this comment

feel free to let me know if it needs further changes, thank you